doc/analyzer.md

```diff
@@ -0,0 +1,53 @@
+# Design: Analyze the Machine Learning Mode in SQLFlow
+
+## Concept
+
+Although the machine learning model is widely used in many fields, it remains mostly a black box. [SHAP](https://github.com/slundberg/shap) is widely used by data scientists to explain the output of any machine learning model.
+
+This design doc introduces how to support the `Analyze SQL` in SQLFlow with SHAP as the backend and display the visualization image to the user.
+
+## User interface
+
+Users usually use a **TRAIN SQL** to train a model and then analyze the model using an **ANALYZE SQL**, the simple pipeline like:
+
+Train SQL:
+
+``` sql
+SELECT * FROM train_table
+TRAIN xgboost.Estimator
+WITH
+    train.objective = "reg:linear"
+COLUMN x
+LABEL y
+INTO my_model;
+```
+
+Analyze SQL:
+
+``` sql
+SELECT * FROM train_table
+ANALYZE my_dnn_model
+WITH
+  plots = force 
+USING TreeExplainer
+```
+
+where:
+- `train_table` is the table of training data.
+- `my_dnn_model` is the trained model.
+- `force` and `summary` is the visualized method.
+- `TreeExplainer` is the [explain type](https://github.com/slundberg/shap#sample-notebooks).
+
+The **Analyze SQL** would display the visualization image on Jupyter like:
+<img src="https://raw.githubusercontent.com/slundberg/shap/master/docs/artwork/boston_dataset.png">
+
+## Implement Details
+
+- Enhance the SQLFlow parser to support the `ANALYZE` keyword.
+- implement the `codgen_shap.go` to generate a SHAP Python program to save the visualization image as HTML format.
```

Comment:
Implement the `codgen_shap.go`, which will generate a SHAP Python program. The Python program will print the resulting image in HTML format to stdout. The Go program will receive the HTML text by running [`CombinedOutput`](https://golang.org/pkg/os/exec/#Cmd.CombinedOutput). The Go program then sends the HTML text to the client as a single message.